src/Pgsql/PgsqlDriver.php

```diff
@@ -221,6 +221,32 @@ public function getTableCreate($tables)
         return [];
     }
 
+    /**
+     * Parses the fully qualified or bare name of the database object (table, view, procedure etc.) into object
+     * with fields "schema" and "table" respectively
+     *
+     * @param string $table Name of the database object with schema or without it
+     *
+     * @return object
+     */
+    private function parseIdent($table)
+    {
+        $query = $this->getQuery(1);
+
+        $query->select(['r[1]','r[2]'])
+            ->from("parse_ident(:t) as r")
+            ->bind(':t', $table);
+
+        $qn = $this->setQuery($query)
+            ->loadRow();
+
+        if ($qn[1]) {
+            return (object) ['schema' => $qn[0], 'table' => $qn[1]];
+        }
+
+        return (object) ['schema' => $this->getDefaultSchema(), 'table' => $qn[0]];
```

Comment:
```suggestion
        $query->select(['current_schemas(false))[1]','r[1]','r[2]'])
            ->from("parse_ident(:t) AS r")
            ->bind(':t', $table);

        $qn = $this->setQuery($query)
            ->loadRow();

        if ($qn[2]) {
            return (object) ['schema' => $qn[1], 'table' => $qn[2]];
        }

        return (object) ['schema' => $qn[0], 'table' => $qn[1]];
```
This should work to save us the additional 2nd query.